packages/dds/tree/src/test/events/delegatingEventEmitter.spec.ts

```diff
@@ -0,0 +1,156 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import { DelegatingEventEmitter } from "../../events";
+
+interface TestEvents {
+    open: () => void;
+    close: (error: boolean) => void;
+}
+
+describe("DelegatingEventEmitter", () => {
+    it("emits events", () => {
+        const emitter = DelegatingEventEmitter.create<TestEvents>();
+        let opened = false;
+        emitter.on("open", () => {
+            opened = true;
```

Comment:
Consider checking that the delegate is only being called once per `emit` call.
```suggestion
            assert(!opened);
            opened = true;
```

Reply:
Good idea!